src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Tree/Dependencies/Snapshot/DependencyTests.cs

```diff
@@ -191,7 +191,8 @@ public void Dependency_EqualsAndGetHashCode()
 
             Assert.Equal(dependency1, dependency2);
             Assert.NotEqual(dependency1, dependency3);
-            Assert.Equal(@"tfm1\providertype\someid1".GetHashCode(), dependency1.GetHashCode());
```

Comment:
This is basically testing the hash code, by writing its implementation in the test. Hash code is implementation detail, it should be testing other properties of the hash code, such as the fact that two objects that considered equal, match in hash code, and those that don't do not.